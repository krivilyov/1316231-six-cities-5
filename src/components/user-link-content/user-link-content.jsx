import React, {Fragment} from "react";
import PropTypes from "prop-types";
import {getIsAuthorizedStatus, getUserAvatar, getUserEMail} from "../../store/selectors";
import {connect} from "react-redux";

const UserLinkContent = (props) => {
  const {userAvatar, userEMail, isAuthorizedStatus} = props;

  return (
    <Fragment>
      <div className="header__avatar-wrapper user__avatar-wrapper"
        style={isAuthorizedStatus ? {backgroundImage: `url(${userAvatar})`} : undefined}
      />
      {isAuthorizedStatus ? (
        <span className="header__user-name user__name">{userEMail}</span>
      ) : (
        <span className="header__login">Sign in</span>
      )}
    </Fragment>
  );
};

UserLinkContent.propTypes = {
  userEMail: PropTypes.string.isRequired,
  userAvatar: PropTypes.string.isRequired,
  isAuthorizedStatus: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  userEMail: getUserEMail(state),
  userAvatar: getUserAvatar(state),
  isAuthorizedStatus: getIsAuthorizedStatus(state),
});

export {UserLinkContent};
export default connect(mapStateToProps)(UserLinkContent);
