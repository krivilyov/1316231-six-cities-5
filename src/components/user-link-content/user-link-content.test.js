import React from "react";
import renderer from "react-test-renderer";
import {UserLinkContent} from "./user-link-content";

describe(`UserLinkContent should render correctly`, () => {
  it(`UserLinkContent should render correctly when isAuthorizedStatus={false}`, () => {
    const tree = renderer
      .create(
          <UserLinkContent
            userEMail={``}
            userAvatar={``}
            isAuthorizedStatus={false}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`UserLinkContent should render correctly when isAuthorizedStatus={true}`, () => {
    const tree = renderer
      .create(
          <UserLinkContent
            userEMail={`user@mail.com`}
            userAvatar={`avatar-url`}
            isAuthorizedStatus={true}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
