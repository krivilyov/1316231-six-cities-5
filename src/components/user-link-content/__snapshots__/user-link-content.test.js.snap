// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLinkContent should render correctly UserLinkContent should render correctly when isAuthorizedStatus={false} 1`] = `
Array [
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  />,
  <span
    className="header__login"
  >
    Sign in
  </span>,
]
`;

exports[`UserLinkContent should render correctly UserLinkContent should render correctly when isAuthorizedStatus={true} 1`] = `
Array [
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
    style={
      Object {
        "backgroundImage": "url(avatar-url)",
      }
    }
  />,
  <span
    className="header__user-name user__name"
  >
    user@mail.com
  </span>,
]
`;
